ant/src/process.lua

```diff
@@ -0,0 +1,175 @@
+-- Adjust package.path to include the current directory
+-- utils
+local json = require(".json")
+local utils = require(".utils")
+local errors = require(".errors")
+local constants = require(".constants")
+
+-- core modules
+local balances = require(".balances")
+local initialize = require(".initialize")
+local records = require(".records")
+local controllers = require(".controllers")
+
+Name = Name or "Arweave Name Token"
+Ticker = Ticker or "ANT"
+Logo = Logo or "LOGO"
+Denomination = Denomination or 1
+
+local ActionMap = {
+	-- write
+	SetController = "SetController",
+	RemoveController = "RemoveController",
+	SetRecord = "SetRecord",
+	RemoveRecord = "RemoveRecord",
+	SetName = "SetName",
+	SetTicker = "SetTicker",
+	--- initialization method for bootstrapping the contract from other platforms ---
+	InitializeState = "InitializeState",
+	-- read
+	GetControllers = "GetControllers",
+	GetRecord = "GetRecord",
+	GetRecords = "GetRecords",
+}
+
+local TokenSpecActionMap = {
+	Info = "Info",
+	Balances = "Balances",
+	Balance = "Balance",
+	Mint = "Mint",
+	Transfer = "Transfer",
+	CreditNotice = "Credit-Notice",
+}
+
+-- Handlers for contract functions
+-- TokenSpecActionMap
+Handlers.add(TokenSpecActionMap.Transfer, utils.hasMatchingTag("Action", TokenSpecActionMap.Transfer), function(msg)
+	local recipient = msg.Tags.Recipient
+	local function checkAssertions()
+		assert(recipient, "Recipient is required")
+		assert(utils.validateArweaveId(recipient), "Invalid recipient")
+		assert(Balances[msg.From], "Sender is not the owner")
+	end
+
+	local inputStatus, inputResult = pcall(checkAssertions)
+
+	if not inputStatus then
+		ao.send({
+			Target = msg.From,
+			Tags = { Error = "Input-Error" },
+			Data = tostring(inputResult),
+		})
+		return
+	end
+	local transferStatus, transferResult = pcall(balances.transfer, recipient)
+
+	if not transferStatus then
+		ao.send({
+			Target = msg.From,
+			Tags = { Error = "Transfer-Error" },
+			Data = tostring(transferResult),
+		})
+		return
+	else
+		ao.send({
+			Target = recipient,
+			Tags = { Action = TokenSpecActionMap.CreditNotice },
+			Data = "Transfer successful",
+		})
+	end
+end)
+
+Handlers.add(TokenSpecActionMap.Balance, utils.hasMatchingTag("Action", TokenSpecActionMap.Balance), function(msg)
+	balances.balance(msg.Tags.Address or msg.From)
+end)
+
+Handlers.add(TokenSpecActionMap.Balances, utils.hasMatchingTag("Action", TokenSpecActionMap.Balances), function(msg)
+	balances.balances()
+end)
+
+Handlers.add(TokenSpecActionMap.Mint, utils.hasMatchingTag("Action", TokenSpecActionMap.Mint), function(msg)
```

Comment:
I dont think we should include this in the spec.  Was there a reason we needed this now?

Reply:
Its just a placeholder to have parity with the token spec - its not currently implemented. Its not in the way or overlapping, figured I would add it as a reminder of "heres all the token handlers from the token spec"

Reply:
to clarify, its implemented to respond with "not implemented" as an error. No minting actually occurs.